### ISSUE TITLE
Does not work with Android P / Pie / 9.0

### ISSUE THREAD

**Issue by dabugen** [Magisk-Modules-Repo/Universal-GMS-Doze#1]
Thanks for the efforts, but this ain´t working with Android 9.0. Google Play Services cannot be forced into doze anymore as it seems - while I can select the "optimize" option on it after applying your module via Magisk, it snaps back to not optimized right afterward. Can you fix this or are the doze times over with Android 9? ;-)

**Comment by gloeyisk** [comment]
Hello and many thanks for used/tested my module and the report.
First of all can you tell me: 
• Which Magisk build you use
• Is that GSI's Pie ROM (tell any specific of the ROM)
• Is my module installed correctly
• Proper logs needed
Anyway, I'm so sorry for any inconvenience caused.

**Comment by dabugen** [comment]
Hello,

thanks for replying :-)

- I am using Magisk 17.2

- It´s Liquid Remix 10.0 (Android 9) over from: https://forum.xda-developers.com/mi-a1/development/rom-liquidremix-t3843247

- Yes, the module is installed correctly and once it is installed, I can even select "Optimized" for "Google Play Services" in battery settings - however, it does not stick, as soon as I selected "Optimized", it jumps back to not optimized. However, without your module, I cannot even select "Optimized" for "Google Play Services", so your module is doing something, but it seems to be Android 9.0 that still is preventing optimization, not your fault for sure, but maybe you can find a fix to make it work with Android 9.0?

Thanks so much :-)

**Comment by gloeyisk** [comment]
Again thanks for the report, I'll take a look when I've free time and searching workarounds for any Pie ROMs.

**Comment by gloeyisk** [comment]
Can test this beta build and report once again? Thanks.
If you've Telegram account you can text me to @gloeyisk.
[univgmsdoze_v1.2.1-beta.zip](https://github.com/Magisk-Modules-Repo/Universal-GMS-Doze/files/2410749/univgmsdoze_v1.2.1-beta.zip)

**Comment by dabugen** [comment]
Thanks a lot, I´ve tried it, but "Optimise" still does not stick for Google Play Services. I´ve made a video for you so that you can see the issue:

https://youtu.be/QwFY3XgJO3o

**Comment by dabugen** [comment]
Yes, odd, could be a Android 9.0 thing in that it cannot be optimized anymore or something, super odd. Thans for your help my friend :)

**Comment by NateDev100** [comment]
Trying this on OnePlus 6 OOS Pie Open Beta 4, it doesn't work, and causes the phone to be unable to go into deep sleep constantly, and the phone is kept awake with screen off all the time.

Removing the module and rebooting twice instantly fixes the issue. Verified multiple times.

**Comment by gloeyisk** [comment]
Hello and thanks for the report.
As I said before this module currently doesn't work properly in any Pie ROMs.
Will try to fix every issues as soon as possible.
